simulation/parser.go

```diff
@@ -143,24 +158,38 @@ func parseAdversaryConfig(adversaryDelays *string, adversaryTypes *string, adver
 	if *adversaryMana != "" {
 		config.AdversaryMana = parseStrToFloat64(*adversaryMana)
 	}
-
-	// no adversary if simulation target is not DS
-	if config.SimulationTarget != "DS" {
-		config.AdversaryTypes = []int{}
+	if *adversaryNodeCounts != "" {
+		config.AdversaryNodeCounts = parseStrToInt(*adversaryNodeCounts)
+	}
+	if *adversaryInitColors != "" {
+		config.AdversaryInitColors = parseStr(*adversaryInitColors)
 	}
 
-	// maximum 3 adversary types allowed for one simulation, one per color
-	if len(config.AdversaryTypes) > 3 {
+	// no adversary if colors are not provided
+	if len(config.AdversaryInitColors) != len(config.AdversaryTypes) {
 		config.AdversaryTypes = []int{}
 	}
 
-	// make sure mana and delays are only defined when adversary type is provided and have the same length
+	// make sure mana, nodeCounts and delays are only defined when adversary type is provided and have the same length
 	if len(config.AdversaryDelays) != 0 && len(config.AdversaryDelays) != len(config.AdversaryTypes) {
 		config.AdversaryDelays = []int{}
 	}
 	if len(config.AdversaryMana) != 0 && len(config.AdversaryMana) != len(config.AdversaryTypes) {
 		config.AdversaryMana = []float64{}
 	}
+	if len(config.AdversaryNodeCounts) != 0 && len(config.AdversaryNodeCounts) != len(config.AdversaryTypes) {
+		config.AdversaryNodeCounts = []int{}
```

Comment:
Might be good to add some Info-level logs to inform the user the setting might be wrong, or the user might get unexpected results due to wrong settings.